### ISSUE TITLE
Infinite loop with `Layout/RedundantLineBreak` and `Style/SingleLineDoEndBlock`

### ISSUE THREAD

**Issue by rickselby** [rubocop/rubocop#12597]
## Expected behavior

Rubocop fixes the cops correctly

## Actual behavior

> Infinite loop detected in code.rb and caused by Layout/RedundantLineBreak -> Style/BlockDelimiters, Style/SingleLineDoEndBlock -> Layout/BlockAlignment, Layout/IndentationWidth, Layout/RedundantLineBreak, Layout/TrailingWhitespace, Style/BlockDelimiters -> Layout/IndentationWidth, Layout/RedundantLineBreak -> Layout/IndentationWidth, Layout/RedundantLineBreak

## Steps to reproduce the problem

`.rubocop.yml`
```
Layout/RedundantLineBreak:
  Enabled: true
  InspectBlocks: true

Style/SingleLineDoEndBlock:
  Enabled: true
```

`code.rb`
```
def foo
  a.each do |x|
    x.each do |y|
      bar y
    end
  end
end
```

## RuboCop version

```
$ rubocop -V
1.59.0 (using Parser 3.3.0.0, rubocop-ast 1.30.0, running on ruby 3.3.0) [x86_64-linux]
```

**Comment by KessaPassa** [comment]
[Layout/RedundantLineBreak InspectBlocks: true](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/RedundantLineBreak)
```rb
# bad
foo(a) do |x|
  puts x
end

# good
foo(a) { |x| puts x }
```

[Style/SingleLineDoEndBlock](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/SingleLineDoEndBlock)
```rb
# bad
foo do |arg| bar(arg) end

# good
foo do |arg|
  bar(arg)
end
```

An infinite loop is occurring as these two cop each other.
Therefore, rubocop cannot decide what the final code will be.

In my opinion, Style/SingleLineDoEndBlock should be preferred because RubyStyle Guide has a description of Style/SingleLineDoEndBlock.
https://rubystyle.guide/#single-line-do-end-block

**Comment by bkuhlmann** [comment]
Hey. :wave: I'm running into this problem too. I'm using the following in case it helps (nearly identical to what is shown above):

**RuboCop Configuration**

*Same as shown above.*

**Code**

``` ruby
demo do |value| puts value end
```

**Command Line**

``` 
rubocop --autocorrect code.rb
```

**Result**

> Infinite loop detected in code.rb and caused by Style/BlockDelimiters, Style/SingleLineDoEndBlock -> Layout/IndentationWidth, Layout/RedundantLineBreak, Layout/TrailingWhitespace, Style/BlockDelimiters -> Layout/RedundantLineBreak

**RuboCop Version**

``` 
1.64.1 (using Parser 3.3.3.0, rubocop-ast 1.31.3, running on ruby 3.3.3) [arm64-darwin23.5.0]
  - rubocop-capybara 2.21.0
  - rubocop-packaging 0.5.2
  - rubocop-performance 1.21.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.0.1
  - rubocop-thread_safety 0.5.1
```

**Comment by mockdeep** [comment]
We're running into this issue as well. It can be reproduced with a basic block:

```rb
a do
 b
end
```

**Comment by dvandersluis** [comment]
Thank you for the reproduction cases. I have opened #13449.
